src/main/java/namor/template/calculator/Sign.java

```diff
@@ -0,0 +1,26 @@
+package namor.template.calculator;
+
+public enum Sign {
+    ADDITION("+"),
+    SUBTRACTION("-"),
+    MULTIPLICATION("*"),
+    DIVISION("/");
+
+    String key;
+
+    Sign(String key){
+        this.key = key;
+    }
+
+    public String getKey() {
+        return key;
+    }
+
+    public static Sign getValue(String key) {
+        if (ADDITION.getKey().equals(key)) { return Sign.ADDITION; }
+        else if (SUBTRACTION.getKey().equals(key)) { return Sign.SUBTRACTION; }
+        else if (MULTIPLICATION.getKey().equals(key)) { return Sign.MULTIPLICATION; }
+        else if (DIVISION.getKey().equals(key)) { return Sign.DIVISION; }
+        else throw new IllegalArgumentException("Input elements are wrong!");
```

Comment:
You don't need all this if-else tree. Get familiar with foreach loop. Also read the API (methods) of classes you use. Enums have few very useful methods. One of them will help you to go through all constants of enum and reduce all this unreadable tree-like code.